### DISCUSSION TITLE
Is there any shallow state?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#333] [Q&A] [answered]
`createState` and `createMutable` always makes deep nested state. Is there a way to create shallow state?

**Comment by ryansolid** [comment, marked as answer]
I don't have a way currently. Partially around the way my proxy solution works. I do find my self hand creating getter objects from time to time to wrap signals but no dynamic shallow proxy solution as of yet. Curious in the use case.  Solid's state objects do create the signals lazily so in non-tracked contexts they don't create signals. So only the properties used ever get tracked. I'm gathering some scenario where you want to never track some nested property anywhere.

**Reply by zhaoyao91** [reply]
I have  global store object, which hold some model objects. The key -> model should be reactive, but I want to treat the model as a blackbox and have their own config about which keys are reactive.

**Reply by ryansolid** [reply]
Hmm.. is it a performance concern? Or a behavior one? As I mentioned signals are created lazily.

I do have a cheater way to accomplish this but it's sort of gaming the system. State only permeates simple objects. If your models were classes then they won't be wrapped. Not sure if that is an adequate work around.

**Reply by zhaoyao91** [reply]
> State only permeates simple objects. If your models were classes then they won't be wrapped.

If it is such a case, that's great and satisify our needs.

To be clarified, we have such a store:

```
const store = createMutable({
  someId: '123',
  someName: 'Bob',
  models: {
    modelA: new ModelA(),
    modelB: new ModelB(),
  }
})
```

As doc said, the state will be deeply proxied to be reactive, but I do not want it to do anything to the modelA and modelB, since I want to control and keep their designed behaviors.

**Reply by ryansolid** [reply]
That should work.. it used to mentioned on the main page but it looks like it got refactored out of the Docs.. I use this hueristic since there are some objects I don't want to wrap like HTMLElements, Dates, Regex etc.. So classes can't be state objects themselves.. but they can be held in state so that their addition or removal triggers updates.

**Reply by zhaoyao91** [reply]
that's cool, so our pattern is supported by nature
